src/main/java/com/limechain/storage/offchain/BasicStorage.java

```diff
@@ -0,0 +1,35 @@
+package com.limechain.storage.offchain;
+
+import java.util.Arrays;
+
+/**
+ * A minimal interface for a basic key-value storage.
+ */
+public interface BasicStorage {
+    // TODO: Shouldn't these `key`s be byte[] instead of String? `String` is not a good container for raw `byte[]`s.
```

Comment:
For the purposes of pure storage interfacing, these are nothing more than `byte[]`.

I left them as `String`s since the underlying implementor `OffchainStore` uses our `KVRepository<String, Object>` and the easier way was to just declare the interface using Strings instead of changing a lot of `Strings` to `byte[]`s.  And not to litter the diff too much with side considerations. 

That's a small issue; spotted in priorly existing behaviour. Anyway, I would go on to fix this later.

Reply:
Fixed in b2b3de0.